### PR TITLE
Codabench connection 

### PR DESCRIPTION
**Pull request by ckcollab**
# A brief description of the purpose of the changes contained in this PR.
Fixes Connection issues with Codabench

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [ ] Code review by reviewer
- [ ] Hand tested by reviewer
- [ ] Ready to merge

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -13,8 +13,8 @@ services:
       - ./src/staticfiles:/var/www/django/static
       - ./certs/caddy:/etc/caddycerts
     ports:
-      - 80:80
-      - 443:443
+      - 80:81
```
Comment (ckcollab) [inline docker-compose.yml]: use environment variables for this! you can do `$DJANGO_PORT` or some such

File: requirements.txt [hunk 2]
```diff
@@ -29,10 +29,11 @@ django-debug-toolbar==1.9.1
 asgiref==2.3.2
 selenium==3.141.0
 Pillow==5.0.0
-django-storages[azure,google]==1.7.1
+django-storages[azure,google]
```
Comment (ckcollab) [inline requirements.txt]: NEED to peg this to a version

File: src/apps/api/serializers/competitions.py [hunk 3]
```diff
@@ -81,7 +81,7 @@ def validate(self, attrs):
 
 class SubmissionSerializer(serializers.ModelSerializer):
     competition = serializers.IntegerField(write_only=True, allow_null=True)
-    phase_index = serializers.IntegerField(write_only=True, allow_null=True)
+    phase_id = serializers.IntegerField(write_only=True, allow_null=True)
```
Comment (ckcollab) [inline src/apps/api/serializers/competitions.py]: revert to phase index

File: requirements.txt [hunk 4]
```diff
@@ -14,11 +14,11 @@ elasticsearch-dsl==7.0.0
 dj-database-url==0.4.2
 social-auth-core==1.5.0
 social-auth-app-django==2.0.0
-django-extensions==1.9.0
+django-extensions==2.2.3
 channels==2.1.2
 #daphne==2.2.2
 gunicorn==19.9.0
-uvicorn==0.8.4
+uvicorn>=0.11.7
```
Comment (ckcollab) [inline requirements.txt]: we should do `==`

File: requirements.txt [hunk 5]
```diff
@@ -28,11 +28,12 @@ tqdm==4.19.5
 django-debug-toolbar==1.9.1
 asgiref==2.3.2
 selenium==3.141.0
-Pillow==5.0.0
-django-storages[azure,google]==1.7.1
+pillow>=7.1.0
```
Comment (ckcollab) [inline requirements.txt]: same here

File: requirements.txt (hunk 4 above)
Comment (Logan-Ruf) [inline requirements.txt:21]:
```suggestion
uvicorn==0.11.7
```

**Pull request merged**